Formula/fish.rb

```diff
@@ -20,12 +20,34 @@ class Fish < Formula
 
   depends_on "pcre2"
 
+  skip_clean "share/fish"
+
   def install
     system "autoconf" if build.head? || build.devel?
+    args = %W[
+      --prefix=#{prefix}
+      --with-extra-functionsdir=#{HOMEBREW_PREFIX}/share/fish/vendor_functions.d
+      --with-extra-completionsdir=#{HOMEBREW_PREFIX}/share/fish/vendor_completions.d
+      --with-extra-confdir=#{HOMEBREW_PREFIX}/share/fish/vendor_conf.d
+      SED=/usr/bin/sed
+    ]
     # In Homebrew's 'superenv' sed's path will be incompatible, so
     # the correct path is passed into configure here.
-    system "./configure", "--prefix=#{prefix}", "SED=/usr/bin/sed"
+
+    system "./configure", *args
+
+    # Skip extra dirs creation during install phase, we'll create them later with mkpath.
+    inreplace "Makefile" do |c|
+      c.gsub! "$(INSTALL) -m 755 -d $(DESTDIR)$(extra_functionsdir)", ""
+      c.gsub! "$(INSTALL) -m 755 -d $(DESTDIR)$(extra_completionsdir)", ""
+      c.gsub! "$(INSTALL) -m 755 -d $(DESTDIR)$(extra_confdir)", ""
+    end
+
     system "make", "install"
+
+    (share/"fish/vendor_functions.d").mkpath
```

Comment:
You'll probably need to shunt these to `post_install`. Fairly sure `brew` won't bother including them in the bottle if they're empty.